docs/troubleshooting.md

```diff
@@ -0,0 +1,90 @@
+# Common troubleshooting tips
+
+## Module not found errors
+
+This is the most common error rules_js users encounter.
+These problems generally stem from a runtime `require` call of some library which was not declared as a dependency.
+
+Fortunately, these problems are not unique to Bazel.
+As described in https://github.com/aspect-build/rules_js/blob/main/docs/pnpm.md#hoisting, rules_js
+should behave the same way `pnpm` does with `hoist=false`.
+These problems are also reproducible under [Yarn PnP](https://yarnpkg.com/features/pnp) because it
+also relies on correct dependencies.
+
+The Node.js documentation describes the algorithm used:
+https://nodejs.org/api/modules.html#loading-from-node_modules-folders
+
+Since the resolution starts from the callsite, the remedy depends on where the `require` statement appears.
+
+### require appears in your code
+
+This is the case when you write a `config.js` file and pass it to a tool.
+
+> This is the "ideal" way for JavaScript libraries to be configured, because it allows an easy
+> "symmetry" where you `require` a library and declare your dependency on it in the same package.
+
+In this case you should add the runtime dependency to your BUILD file where the `config.js` is a source.
+
+For example,
+
+```starlark
+js_library(
+    name = "requires_foo",
+    srcs = ["config.js"],                  # contains "require('foo')"
+    data = ["//my/pkg:node_modules/foo"],  # satisfies that require
```

Comment:
Where is `my/pkg` relative to `config.js`?

Reply:
It's unspecified. Here I'm trying to suggest you need to figure out where the foo package is linked. Do you have a suggestion how to change this?

Reply:
I think `config.js` needs to be within `my/pkg` for this to work. Is the "satisfies that require" comment enough? Or could change it to just `:node_modules/foo`.

What you have might be fine, I'm not sure what the clearest way of saying it is.

Reply:
got it, I agree just removing `//my/pkg` makes it more typical of what a user will have to do.

Would love to have more examples you've run across included or linked in this doc!